frontend/components/idioms.jsx

```diff
@@ -7,41 +7,41 @@
   Some references used:
   https://claritydev.net/blog/the-most-common-mistakes-when-using-react
 */
-import { useEffect } from 'react'
+import { useEffect } from 'react';
+
+export const ShowButton = ({ buttonText }) => {
+  <button>{buttonText}</button>;
+};
 
 export function FunctionsAsComponents({ buttonText = 'Start Now' }) {
-  const showButton = () => {
-    <button>
-      {buttonText}
-    </button>
-  }
-
-  return (<div>
-    {showButton()}
-  </div>)
+  return (
+    <div>
+      <ShowButton buttonText={buttonText} />
+    </div>
+  );
 }
 
 export function objectCopying() {
-  const object = { a: { c: 1 }, b: 2 }
-  const copy = { ...object }
-  copy.a.c = 2
-  return { ...object }
+  const object = { a: { c: 1 }, b: 2 };
+  const copy = object; // doesn't make sense to deconstruct for nothing
+  copy['a']['c'] = 2; // not sure if this is better though feels the same
+  return object;
 }
 
 export function arrayCopying() {
-  const array = [{ a: 1}, {a: 2}, {a: 3}]
-  const copy = [...array]
+  const array = [{ a: 1 }, { a: 2 }, { a: 3 }];
+  const copy = array; // doesn't make sense to deconstruct for nothing
   return copy;
 }
 
 // user abort
+const fetchData = async (fetchURL) => {
+  // no need to create the function every time on the useEffect
+  await fetch(fetchURL);
+};
 export function UseEffect({ fetchURL, label }) {
   useEffect(() => {
-    const fetchData = async () => {
-      await fetch(fetchURL);
-    };
-    
-    fetchData();
+    fetchData(fetchURL); //not doing much with the data not sure if meaningful
```

Comment:
These proposed changes don't make sense. This is not the issue with the original code. Consider what happens on dismount.